streampipes-extensions/streampipes-sinks-internal-jvm/src/main/java/org/apache/streampipes/sinks/internal/jvm/notification/InternalStreamPipesNotificationSink.java

```diff
@@ -61,63 +53,59 @@ public class NotificationProducer extends StreamPipesDataSink {
 
   @Override
   public void onInvocation(SinkParams parameters, EventSinkRuntimeContext context) throws
-      SpRuntimeException {
-    this.title = parameters.extractor().singleValueParameter(TITLE_KEY, String.class);
-    this.content = parameters.extractor().singleValueParameter(CONTENT_KEY, String.class);
-    this.silentPeriodInSeconds = parameters.extractor().singleValueParameter(SILENT_PERIOD, Integer.class) * 60;
-    this.correspondingPipelineId = parameters.getModel().getCorrespondingPipeline();
-    this.correspondingUser = parameters.getModel().getCorrespondingUser();
+                                                                                   SpRuntimeException {
+    super.onInvocation(parameters, context);
+    this.title = parameters.extractor()
+                           .singleValueParameter(TITLE_KEY, String.class);
+    this.content = parameters.extractor()
+                             .singleValueParameter(CONTENT_KEY, String.class);
+    this.correspondingPipelineId = parameters.getModel()
+                                             .getCorrespondingPipeline();
+    this.correspondingUser = parameters.getModel()
+                                       .getCorrespondingUser();
     this.client = context.getStreamPipesClient();
   }
 
   @Override
-  public void onEvent(Event inputEvent) {
-    if (shouldSendNotification()) {
-      Date currentDate = new Date();
-      Notification notification = new Notification();
-      notification.setId(UUID.randomUUID().toString());
-      notification.setRead(false);
-      notification.setTitle(title);
-      notification.setMessage(PlaceholderExtractor.replacePlaceholders(inputEvent, content));
-      notification.setCreatedAt(currentDate);
-      notification.setCreatedAtTimestamp(currentDate.getTime());
-      notification.setCorrespondingPipelineId(correspondingPipelineId);
-      notification.setTargetedAt(correspondingUser);
-
-      // TODO add targeted user to notification object
-
-      client.notificationsApi().add(notification);
-      this.lastMessageEpochSecond = Instant.now().getEpochSecond();
-    }
+  public void onNotificationEvent(Event inputEvent) {
+
+    Date currentDate = new Date();
+    Notification notification = new Notification();
+    notification.setId(UUID.randomUUID()
+                           .toString());
+    notification.setRead(false);
+    notification.setTitle(title);
+    notification.setMessage(PlaceholderExtractor.replacePlaceholders(inputEvent, content));
+    notification.setCreatedAt(currentDate);
+    notification.setCreatedAtTimestamp(currentDate.getTime());
+    notification.setCorrespondingPipelineId(correspondingPipelineId);
+    notification.setTargetedAt(correspondingUser);
+
+    // TODO add targeted user to notification object
```

Comment:
Can we remove this and create an issue for this feature instead?

Reply:
Sure, here it is https://github.com/apache/streampipes/issues/2314